### PR TITLE
feat: implement bis api for inscription by address query

### PR DESCRIPTION
**Pull request by alter-eggo**
Inscription by address query  is used for filtering native segwit utxos.
This pr changes the the source of data to bis api, other queries will be refactored later

### CODE REVIEW

File: packages/models/src/crypto-assets/bitcoin/inscription.model.ts [hunk 1]
```diff
@@ -156,3 +157,82 @@ export function createInscription(inscription: RawInscription): Inscription {
...
+    offset,
+    address: inscription.owner_wallet_address,
+    preview,
+    title,
+    genesisBlockHeight: inscription.genesis_height,
+
+    // TODO: currently no data from BestinSlot API
+    genesisBlockHash: '0',
+    genesisTimestamp: 0,
+    value: '0',
```
Comment (alter-eggo) [inline packages/models/src/crypto-assets/bitcoin/inscription.model.ts:185]: wonder if we need this info? 🤔 it doesn't come from bis api, should we ask to fix it?

File: packages/models/src/inscriptions/inscriptions.model.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+export interface BestinslotInscriptionResponse {
```
Comment (fbwoolf) [inline packages/models/src/inscriptions/inscriptions.model.ts:1]: I think we were making an effort to keep api models with the queries and only use the models pkg for the models we map/create/use in the app ...like a DTO pattern?

**Review by fbwoolf — APPROVED** [review]

File: packages/models/src/inscriptions/inscriptions.model.ts (hunk 2 above)
Comment (kyranjamie) [reply packages/models/src/inscriptions/inscriptions.model.ts:1]: +1, this is specific to the API so belongs with the fetcher/query. 

`@leather.io/models` shouldn't "know" about BIS.

File: packages/query/src/bitcoin/bitcoin-client.ts [hunk 3]
```diff
@@ -156,6 +137,23 @@ interface RunesOutputsByAddressResponse {
   data: RunesOutputsByAddress[];
 }
 
+interface BestinSlotInscriptionByAddressArgs {
```
Comment (kyranjamie) [inline packages/query/src/bitcoin/bitcoin-client.ts:140]:
```suggestion
interface BestInSlotInscriptionByAddressArgs {
```

**Review by kyranjamie — APPROVED** [review]
LGTM. I think it'd be better to use zod + type inference to get the type though, this way we can validate the schema and ensure it's as we expect

**Comment by alter-eggo** [comment]
@kyranjamie added zod validation, will double check it, once bis team send type definitions

File: packages/query/src/bitcoin/ordinals/inscriptions.query.ts [hunk 4]
```diff
@@ -208,33 +212,38 @@ export function useGetInscriptionsInfiniteQuery({
...
+      return lastPage.offset + bestinslotInscriptionsRequestNum;
     },
-    refetchOnMount: false,
-    refetchOnReconnect: false,
-    refetchOnWindowFocus: false,
-    staleTime: 3 * 60 * 1000,
+    // refetchOnMount: false,
+    // refetchOnReconnect: false,
+    // refetchOnWindowFocus: false,
+    // staleTime: 3 * 60 * 1000,
```
Comment (fbwoolf) [inline packages/query/src/bitcoin/ordinals/inscriptions.query.ts:246]: Did you mean to leave these commented out?

Comment (alter-eggo) [reply packages/query/src/bitcoin/ordinals/inscriptions.query.ts:246]: yes, I've reverted it in main pr

**Pull request merged**